crates/oxc_codegen/src/sourcemap_builder.rs

```diff
@@ -148,6 +157,55 @@ impl SourcemapBuilder {
         (original_line as u32, original_column)
     }
 
+    fn search_original_line(&mut self, position: u32) -> usize {
+        let lines = &self.line_offset_tables.lines;
+        let idx = self.last_line_lookup;
+
+        if position >= lines[idx].byte_offset_to_start_of_line {
+            self.search_original_line_forwards(position)
+        } else {
+            self.search_original_line_backwards(position)
+        }
+    }
+
+    fn search_original_line_forwards(&mut self, position: u32) -> usize {
+        let lines = &self.line_offset_tables.lines;
+        let mut idx = self.last_line_lookup;
+
+        let cap = min(idx + 16, lines.len() - 1);
+        idx = min(idx + 1, cap);
```

Comment:
The bug is on this line. You start searching at `idx + 17`, so most of the time you miss the answer and it falls through to binary search.

I'm not sure what the fix is. `max` not `min`? Not clear to me what the purpose of this line is.